pkg/executables/kubectl.go

```diff
@@ -825,3 +825,116 @@ func (k *Kubectl) GetBundles(ctx context.Context, kubeconfigFile, name, namespac
 
 	return response, nil
 }
+
+func (k *Kubectl) PatchTolerations(ctx context.Context, taints []corev1.Taint, resource string, name string, kubeconfigFile string, namespace string, path string) error {
+	if len(taints) > 0 {
+		var tolerations []string
+		format := "{\"op\": \"add\", \"path\": \"%s\", \"value\":{\"key\":\"%s\",\"operator\":\"Equal\",\"value\":\"%s\",\"effect\":\"%s\"}}"
+		for _, taint := range taints {
+			if taint.Effect != "NoExecute" {
+				tolerations = append(tolerations, fmt.Sprintf(format, path, taint.Key, taint.Value, taint.Effect))
+			}
+		}
+		if len(tolerations) > 0 {
+			params := []string{
+				"patch", resource, name,
+				"--type=json", fmt.Sprintf("-p=[%s]", strings.Join(tolerations, ", ")), "-n", namespace, "--kubeconfig", kubeconfigFile,
+			}
+			_, err := k.executable.Execute(ctx, params...)
+			if err != nil {
+				return err
+			}
+		}
+	}
+	return nil
+}
+
+type Toleration struct {
+	Effect   string
+	Key      string
+	Operator string
+	Value    string
+}
+
+func (k *Kubectl) UpdateTolerations(ctx context.Context, oldTaints []corev1.Taint, newTaints []corev1.Taint, resource string, name string, kubeconfigFile string, namespace string, path string) error {
+	params := []string{
+		"get", resource, name,
+		"-o", "jsonpath={range .spec.template.spec.tolerations[*]}[{@}] {end}",
+		"-n", namespace, "--kubeconfig", kubeconfigFile,
+	}
+	output, err := k.executable.Execute(ctx, params...)
+	if err != nil {
+		return err
+	}
+	var appliedTolerations []Toleration
+	tolerationStr := output.String()
+	if len(tolerationStr) > 0 {
+		tolerations := strings.Split(tolerationStr, " ")
+		for _, ent := range tolerations {
+			ent := strings.TrimPrefix(ent, "[{")
+			ent = strings.TrimSuffix(ent, "}]")
+			comps := strings.Split(ent, ",")
+			valid := false
+			var toleration Toleration
+			for _, comp := range comps {
+				if len(comp) == 0 {
+					continue
+				}
+				parts := strings.Split(comp, ":")
+				parts[0] = strings.TrimPrefix(parts[0], "\"")
+				parts[0] = strings.TrimSuffix(parts[0], "\"")
+				parts[1] = strings.TrimPrefix(parts[1], "\"")
+				parts[1] = strings.TrimSuffix(parts[1], "\"")
+				if parts[0] == "effect" {
+					toleration.Effect = parts[1]
+					valid = true
+				}
+				if parts[0] == "key" {
+					toleration.Key = parts[1]
+					valid = true
+				}
+				if parts[0] == "operator" {
+					toleration.Operator = parts[1]
+					valid = true
+				}
+				if parts[0] == "value" {
+					toleration.Value = parts[1]
+					valid = true
+				}
+			}
+			if valid {
+				appliedTolerations = append(appliedTolerations, toleration)
+			}
+		}
+	}
+	var finalTolerations []string
+	format := "{\"key\":\"%s\",\"operator\":\"%s\",\"value\":\"%s\",\"effect\":\"%s\"}"
+	for _, toleration := range appliedTolerations {
```

Comment:
What do you think about making a hashmap with `oldTaints` so you don't have two nested loops?
In fact, you only need a set, bc you just want to check for presence
You might be able to use `types.Lookup`